af04f855-58c4-4db3-a191-45fe33381679/scripts/actions.py

```diff
@@ -799,3 +795,22 @@ def CheckTournamentValid(group, x=0, y=0):
 			notify("{}'s deck is legal in a FFG + Apoka format".format(me))
 	else:
 		notify("{}'s deck does contain some invalid cards for the selected format".format(me))
+
+#------------------------------------------------------------------------------
+# Choose secret Number
+#------------------------------------------------------------------------------
+
+chosenNumber = None
+def askSecretNumber(group, x=0, y=0):
+	global chosenNumber
+	chosenNumber = askInteger("Choose a secret number", 0)
+	if chosenNumber != None:
+		notify("{} has chosen a secret number".format(me))
+	return
+
+def revealSecretNumber(group, x=0, y=0):
+	if chosenNumber != None:
+		notify("{} has chosen {} as a secret number".format(me, chosenNumber))
+	else:
+		notify("{} doesn't have any secret number chosen".format(chosenNumber))
```

Comment:
Format should be getting 'me' instead of chosenNumber

Instead of notify I think you could use whisper to message only the player triggering the action, to say
'You have not chosen a secret number'

Reply:
Should be done with the next commit =p